system/jlib/jmetrics.cpp

```diff
@@ -138,35 +139,49 @@ bool MetricsManager::addMetric(const std::shared_ptr<IMetric> &pMetric)
     bool rc = false;
     std::unique_lock<std::mutex> lock(metricVectorMutex);
     std::string name = pMetric->queryName();
-    auto metaData = pMetric->queryMetaData();
-    for (auto &metaDataIt: metaData)
-    {
-        name.append(".").append(metaDataIt.value);
-    }
+    std::regex pattern("^[A-Za-z][A-Za-z0-9.]*[A-Za-z0-9]$");
```

Comment:
trivial: This should be a member of MetricManager, so that the pattern object can be created once (which can be expensive), and reused by all calls.  If this is going to be called for large numbers it may become significant.